### ISSUE TITLE
[bug] "It appears we are experienceing some issues" on fresh install

### ISSUE THREAD

**Issue by evanrich** [siderolabs/omni#463]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

UI loads the background and top header, but otherwise shows something in the middle saying
![image](https://github.com/user-attachments/assets/b521f26d-45f2-404b-8d31-f94549224ea8)

I've tried both insecure/secure modes, and chrome/edge, can't get it to work either way.  I've also tried using the recommended action her:
```
Flag --siderolink-api-bind-addr has been deprecated, --deprecated, use --machine-api-bind-addr
```
but that didn't change anything either.  a scan of the logs indicate no errors.

### Expected Behavior

it loads

### Steps To Reproduce

follow the on-prem guide, and launch any version (0.31.1, 0.38.0, latest, etc) with the folllowing:
```
docker run \
  --net=host \
  --privileged \
  --cap-add=NET_ADMIN \
  -v $PWD/etcd:/_out/etcd \
  -v $PWD/omni.asc:/omni.asc \
  ghcr.io/siderolabs/omni:v0.39.0 \
    --account-id=abcd-1234-9876-zyxw \
    --name=omni \
    --private-key-source=file:///omni.asc \
    --event-sink-port=8091 \
    --bind-addr=0.0.0.0:8080 \
    --siderolink-api-bind-addr=0.0.0.0:8090 \
    --k8s-proxy-bind-addr=0.0.0.0:8100 \
    --advertised-api-url=https://omni.tld.io/ \
    --siderolink-api-advertised-url=https://omni.tld.io:8090/ \
    --siderolink-wireguard-advertised-addr=192.168.5.2:50180 \
    --advertised-kubernetes-proxy-url=https://omni.tld.io:8100/ \
    --auth-auth0-enabled=true \
    --auth-auth0-domain=dev-abcd1234.us.auth0.com \
    --auth-auth0-client-id=12345678TOOCOOLFORSCHOOL \
    --initial-users=bezos@bookstore.com
```

I do see one error at the top of the logs:
```
{"level":"warn","ts":1721115594.4964077,"caller":"service/service.go:112","msg":"failed to load state from storage","component":"server","component":"discovery_service","error":"failed to open file: open _out/secondary-storage/discovery-service-state.binpb: no such file or directory"}
```
but thats it.

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### Anything else?

logs:
```
Flag --siderolink-api-bind-addr has been deprecated, --deprecated, use --machine-api-bind-addr
{"level":"info","ts":1721115592.4526026,"caller":"omni/main.go:114","msg":"starting Omni","version":"v0.39.0"}
{"level":"info","ts":1721115592.4529274,"caller":"omni/state_etcd.go:148","msg":"starting embedded etcd server","component":"server","component":"embedded_etcd","data_dir":"_out/etcd/"}
{"level":"warn","ts":1721115592.4530108,"caller":"embed/config.go:687","msg":"Running http and grpc server on single port. This is not recommended for production.","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115592.453032,"caller":"embed/etcd.go:127","msg":"configuring peer listeners","component":"server","component":"embedded_etcd","listen-peer-urls":["http://localhost:0"]}
{"level":"info","ts":1721115592.4536762,"caller":"embed/etcd.go:135","msg":"configuring client listeners","component":"server","component":"embedded_etcd","listen-client-urls":["http://localhost:2379"]}
{"level":"info","ts":1721115592.45389,"caller":"embed/etcd.go:309","msg":"starting an etcd server","component":"server","component":"embedded_etcd","etcd-version":"3.5.14","git-sha":"Not provided (use ./build instead of go build)","go-version":"go1.22.5","go-os":"linux","go-arch":"amd64","max-cpu-set":8,"max-cpu-available":8,"member-initialized":true,"name":"default","data-dir":"_out/etcd/","wal-dir":"","wal-dir-dedicated":"","member-dir":"_out/etcd/member","force-new-cluster":false,"heartbeat-interval":"100ms","election-timeout":"1s","initial-election-tick-advance":true,"snapshot-count":100000,"max-wals":5,"max-snapshots":5,"snapshot-catchup-entries":5000,"initial-advertise-peer-urls":["http://localhost:2380"],"listen-peer-urls":["http://localhost:0"],"advertise-client-urls":["http://localhost:2379"],"listen-client-urls":["http://localhost:2379"],"listen-metrics-urls":[],"cors":["*"],"host-whitelist":["*"],"initial-cluster":"","initial-cluster-state":"new","initial-cluster-token":"","quota-backend-bytes":2147483648,"max-request-bytes":1572864,"max-concurrent-streams":4294967295,"pre-vote":true,"initial-corrupt-check":true,"corrupt-check-time-interval":"0s","compact-check-time-enabled":true,"compact-check-time-interval":60,"auto-compaction-mode":"periodic","auto-compaction-retention":18000,"auto-compaction-interval":"5h0m0s","discovery-url":"","discovery-proxy":"","downgrade-check-interval":"5s"}
{"level":"warn","ts":1721115592.453973,"caller":"fileutil/fileutil.go:53","msg":"check file permission","component":"server","component":"embedded_etcd","error":"directory \"_out/etcd/\" exist, but the permission is \"drwxr-xr-x\". The recommended permission is \"-rwx------\" to prevent possible unprivileged access to the data"}
{"level":"info","ts":1721115592.4650388,"caller":"etcdserver/backend.go:81","msg":"opened backend db","component":"server","component":"embedded_etcd","path":"_out/etcd/member/snap/db","took":0.010753365}
{"level":"info","ts":1721115592.4671576,"caller":"etcdserver/server.go:532","msg":"No snapshot found. Recovering WAL from scratch!","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115592.4690013,"caller":"etcdserver/raft.go:530","msg":"restarting local member","component":"server","component":"embedded_etcd","cluster-id":"cdf818194e3a8c32","local-member-id":"8e9e05c52164694d","commit-index":248}
{"level":"info","ts":1721115592.4691079,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d switched to configuration voters=()","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115592.469155,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became follower at term 2","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115592.469168,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"newRaft 8e9e05c52164694d [peers: [], term: 2, commit: 248, applied: 0, lastindex: 248, lastterm: 2]","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115592.4731114,"caller":"mvcc/kvstore.go:418","msg":"kvstore restored","component":"server","component":"embedded_etcd","current-rev":243}
{"level":"info","ts":1721115592.4763622,"caller":"etcdserver/quota.go:94","msg":"enabled backend quota with default value","component":"server","component":"embedded_etcd","quota-name":"v3-applier","quota-size-bytes":2147483648,"quota-size":"2.1 GB"}
{"level":"info","ts":1721115592.4774268,"caller":"etcdserver/corrupt.go:96","msg":"starting initial corruption check","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d","timeout":7}
{"level":"info","ts":1721115592.4776185,"caller":"etcdserver/corrupt.go:177","msg":"initial corruption checking passed; no corruption","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d"}
{"level":"info","ts":1721115592.4776506,"caller":"etcdserver/server.go:867","msg":"starting etcd server","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d","local-server-version":"3.5.14","cluster-version":"to_be_decided"}
{"level":"info","ts":1721115592.4779377,"caller":"v3rpc/health.go:61","msg":"grpc service status changed","component":"server","component":"embedded_etcd","service":"","status":"SERVING"}
{"level":"info","ts":1721115592.4780474,"caller":"fileutil/purge.go:50","msg":"started to purge file","component":"server","component":"embedded_etcd","dir":"_out/etcd/member/snap","suffix":"snap.db","max":5,"interval":30}
{"level":"info","ts":1721115592.4781244,"caller":"fileutil/purge.go:50","msg":"started to purge file","component":"server","component":"embedded_etcd","dir":"_out/etcd/member/snap","suffix":"snap","max":5,"interval":30}
{"level":"info","ts":1721115592.4781342,"caller":"fileutil/purge.go:50","msg":"started to purge file","component":"server","component":"embedded_etcd","dir":"_out/etcd/member/wal","suffix":"wal","max":5,"interval":30}
{"level":"info","ts":1721115592.4783995,"caller":"etcdserver/server.go:767","msg":"starting initial election tick advance","component":"server","component":"embedded_etcd","election-ticks":10}
{"level":"info","ts":1721115592.4784303,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d switched to configuration voters=(10276657743932975437)","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115592.4784896,"caller":"membership/cluster.go:421","msg":"added member","component":"server","component":"embedded_etcd","cluster-id":"cdf818194e3a8c32","local-member-id":"8e9e05c52164694d","added-peer-id":"8e9e05c52164694d","added-peer-peer-urls":["http://localhost:2380"]}
{"level":"info","ts":1721115592.478607,"caller":"membership/cluster.go:584","msg":"set initial cluster version","component":"server","component":"embedded_etcd","cluster-id":"cdf818194e3a8c32","local-member-id":"8e9e05c52164694d","cluster-version":"3.5"}
{"level":"info","ts":1721115592.4786465,"caller":"api/capability.go:75","msg":"enabled capabilities for version","component":"server","component":"embedded_etcd","cluster-version":"3.5"}
{"level":"info","ts":1721115592.4801803,"caller":"embed/etcd.go:278","msg":"now serving peer/client/metrics","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d","initial-advertise-peer-urls":["http://localhost:2380"],"listen-peer-urls":["http://localhost:0"],"advertise-client-urls":["http://localhost:2379"],"listen-client-urls":["http://localhost:2379"],"listen-metrics-urls":[]}
{"level":"info","ts":1721115592.4801872,"caller":"embed/etcd.go:598","msg":"serving peer traffic","component":"server","component":"embedded_etcd","address":"127.0.0.1:38575"}
{"level":"info","ts":1721115592.480924,"caller":"embed/etcd.go:570","msg":"cmux::serve","component":"server","component":"embedded_etcd","address":"127.0.0.1:38575"}
{"level":"info","ts":1721115594.3701236,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d is starting a new election at term 2","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.3702312,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became pre-candidate at term 2","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.370278,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d received MsgPreVoteResp from 8e9e05c52164694d at term 2","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.3703175,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became candidate at term 3","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.3703377,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d received MsgVoteResp from 8e9e05c52164694d at term 3","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.3703668,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"8e9e05c52164694d became leader at term 3","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.3703887,"logger":"raft","caller":"etcdserver/zap_raft.go:77","msg":"raft.node: 8e9e05c52164694d elected leader 8e9e05c52164694d at term 3","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.375197,"caller":"etcdserver/server.go:2118","msg":"published local member to cluster through raft","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d","local-member-attributes":"{Name:default ClientURLs:[http://localhost:2379]}","request-path":"/0/members/8e9e05c52164694d/attributes","cluster-id":"cdf818194e3a8c32","publish-timeout":7}
{"level":"info","ts":1721115594.37528,"caller":"embed/serve.go:103","msg":"ready to serve client requests","component":"server","component":"embedded_etcd"}
{"level":"info","ts":1721115594.375573,"caller":"v3rpc/health.go:61","msg":"grpc service status changed","component":"server","component":"embedded_etcd","service":"","status":"SERVING"}
{"level":"info","ts":1721115594.376316,"caller":"embed/serve.go:187","msg":"serving client traffic insecurely; this is strongly discouraged!","component":"server","component":"embedded_etcd","traffic":"grpc+http","address":"127.0.0.1:2379"}
{"level":"info","ts":1721115594.3823495,"caller":"omni/state_etcd_election.go:45","msg":"running the etcd election campaign","component":"elections"}
{"level":"info","ts":1721115594.3867624,"caller":"omni/state_etcd_election.go:67","msg":"won the etcd election campaign","component":"elections"}
{"level":"info","ts":1721115594.4216022,"caller":"keyprovider/storage.go:89","msg":"using existing key store","name":"d0cab17d2345fb050250089a09ac0a601b5e6abecbfddc9d2f53a2bc24bfec5b","slot":"Omni (Used for etcd data encryption) how-to-guide@siderolabs.com <7faec5692f35a748788cb2706ed86287468bbd65>","version":1,"key_identity":"Omni (Used for etcd data encryption) how-to-guide@siderolabs.com <7faec5692f35a748788cb2706ed86287468bbd65>"}
{"level":"info","ts":1721115594.4837554,"caller":"user/user.go:33","msg":"some users already exist, skipping initial users"}
.
.
.
.
.

{"level":"info","ts":1721114932.4255135,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.6.3","job":"reconcile","busy":0.145392461}
{"level":"info","ts":1721114932.5685663,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.6.4","job":"reconcile","busy":0.142892271}
{"level":"info","ts":1721114932.7096412,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.5.6","job":"reconcile","busy":0.140926157}
{"level":"info","ts":1721114932.8518562,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.6.5","job":"reconcile","busy":0.141998454}
{"level":"info","ts":1721114932.9946854,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.6.6","job":"reconcile","busy":0.142673254}
{"level":"info","ts":1721114933.141747,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.6.7","job":"reconcile","busy":0.146851398}
{"level":"info","ts":1721114933.2838428,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.7.0","job":"reconcile","busy":0.14190807}
{"level":"info","ts":1721114933.4275038,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.7.1","job":"reconcile","busy":0.14351021}
{"level":"info","ts":1721114933.569267,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.7.2","job":"reconcile","busy":0.141608738}
{"level":"info","ts":1721114933.7124352,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.7.3","job":"reconcile","busy":0.142998883}
{"level":"info","ts":1721114933.8553631,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.7.4","job":"reconcile","busy":0.142766864}
{"level":"info","ts":1721114933.9971154,"caller":"qruntime/qruntime.go:284","msg":"reconcile succeeded","component":"omni_runtime","controller":"TalosExtensionsController","namespace":"default","type":"TalosVersions.omni.sidero.dev","id":"1.7.5","job":"reconcile","busy":0.141587942}
{"level":"info","ts":1721114939.648258,"caller":"logging/handler.go:59","msg":"HTTP request done","component":"server","handler":"static","request_url":"/","method":"GET","remote_addr":"172.69.135.55","duration":0.000085904,"status":304,"response_length":0}
{"level":"info","ts":1721114940.7609951,"caller":"logging/handler.go:59","msg":"HTTP request done","component":"server","handler":"static","request_url":"/","method":"GET","remote_addr":"172.69.135.55","duration":0.00008776,"status":304,"response_length":0}
{"level":"info","ts":1721114954.2584105,"caller":"logging/handler.go:59","msg":"HTTP request done","component":"server","handler":"static","request_url":"/","method":"GET","remote_addr":"172.69.135.55","duration":0.000088401,"status":304,"response_length":0}
{"level":"info","ts":1721114955.5379453,"caller":"logging/handler.go:59","msg":"HTTP request done","component":"server","handler":"static","request_url":"/","method":"GET","remote_addr":"172.69.135.55","duration":0.000089396,"status":304,"response_length":0}
{"level":"info","ts":1721114956.1870675,"caller":"logging/handler.go:59","msg":"HTTP request done","component":"server","handler":"static","request_url":"/","method":"GET","remote_addr":"172.69.135.55","duration":0.000051275,"status":304,"response_length":0}
{"level":"info","ts":1721114980.3881543,"caller":"etcdserver/corrupt.go:276","msg":"starting compact hash check","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d","timeout":7}
{"level":"info","ts":1721114980.3882244,"caller":"etcdserver/corrupt.go:335","msg":"finished compaction hash check","component":"server","component":"embedded_etcd","number-of-hashes-checked":0}
{"level":"info","ts":1721115040.3890312,"caller":"etcdserver/corrupt.go:276","msg":"starting compact hash check","component":"server","component":"embedded_etcd","local-member-id":"8e9e05c52164694d","timeout":7}
{"level":"info","ts":1721115040.389104,"caller":"etcdserver/corrupt.go:335","msg":"finished compaction hash check","component":"server","component":"embedded_etcd","number-of-hashes-checked":0}
```

Update: adding --debug produces no additional useful logging content as to what the issue might be.  I've also tried having someone else setup auth0 to see if it was a problem there, but that did not work either.  I have also tried using auth and no auth, using cert-bot per the instructions, as well as a ingress with TLS in my kubernetes cluster that then pointed at the VM, also did nothing.  Local access (via the machine IP) produces a blank sreen with the omni wallpaper, but no content.

**Comment by Unix4ever** [comment]
We need the frontend logs (F12 -> network) to tell more.
Also if you have the reverse proxy in front of Omni please refer to this guide: https://omni.siderolabs.com/docs/how-to-guides/self-hosted/how-to-expose-omni-with-nginx/

**Comment by evanrich** [comment]
@Unix4ever here you go:
![image](https://github.com/user-attachments/assets/bb204bb1-0b38-4aba-acab-c3112522bf22)
the blacked out part is just domain.io  I'm using traefik in Kubernetes to externally route to this (my ingress is in kubernetes) which for testing sake currently terminates ssl and goes unencrypted to omni:

```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: omni
  namespace: external-services
  annotations:
    kubernetes.io/tls-acme: 'true'
    external-dns.alpha.kubernetes.io/hostname: omni.domain.io
    external-dns.alpha.kubernetes.io/ingress-hostname-source: annotation-only
    external-dns.alpha.kubernetes.io/ttl: "120"
spec:
  ingressClassName: traefik
  tls:
    - hosts:
        - omni.domain.io
      secretName: omni-tls
  rules:
    - host: omni.domain.io
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: omni
                port:
                  number: 8080
```

The service and endpoint are defined with 
```
apiVersion: v1
kind: Service
metadata:
  annotations:
  name: omni
  namespace: external-services
  labels:
    app: omni
spec:
  ports:
  - name: http
    port: 8080
  - name: api
    port: 8090
  - name: proxy
    port: 8100
  type: ClusterIP

---
apiVersion: v1
kind: Endpoints
metadata:
  name: omni
  namespace: external-services
  labels:
    app: omni
subsets:
- addresses:
  - ip: 192.168.5.2
  ports:
  - name: http
    protocol: TCP
    port: 8080
  - name: api
    protocol: TCP
    port: 8090
  - name: proxy
    protocol: TCP
    port: 8100
```
I do this for a number of external services that are running in VMs without issue so I don't think this is the problem

**Comment by evanrich** [comment]
Forgot to test locally, via http://IP:8080
![image](https://github.com/user-attachments/assets/4053734f-6ccf-4e77-926d-fddf120aaba6)

so it looks like there might be an issue running insecure maybe?  (despite the docs saying you can do it by removing the tls options?)

**Comment by evanrich** [comment]
ok I made some progress.  I edited the hosts file on my machine, to point the domain name directly to the IP of the server, so something like:
```
192.168.5.2 omni.domain.io
```
and it works, although initially it had the wrong email.   when I comment out the hosts file entry, it fails again.  This seems like an ingress issue after all, except then how does the auth0 call back work? doesn't that go from Auth0 through my external DNS into the omni instance?
